accounts-db/src/tiered_storage.rs

```diff
@@ -302,4 +306,137 @@ mod tests {
+    /// The created test account might have default rent_epoch
+    /// and write_version.
+    fn create_test_account(seed: u64) -> (StoredMeta, AccountSharedData) {
+        let data_byte = (seed % 256) as u8;
+        let account = Account {
+            lamports: seed,
+            data: (0..seed as usize).map(|_| data_byte).collect(),
+            owner: Pubkey::new_unique(),
+            executable: seed % 2 > 0,
+            rent_epoch: if seed % 3 > 0 { seed } else { u64::MAX },
+        };
+
+        let stored_meta = StoredMeta {
+            write_version_obsolete: if seed % 5 > 0 { seed } else { u64::MAX },
+            pubkey: Pubkey::new_unique(),
+            data_len: seed,
+        };
+        (stored_meta, AccountSharedData::from(account))
+    }
+
+    /// The helper function for all write_accounts tests.
+    /// Currently only supports hot accounts.
+    fn write_accounts_test_impl(
+        path_suffix: &str,
+        account_data_sizes: &[u64],
+        format: TieredStorageFormat,
+    ) {
+        let accounts: Vec<(StoredMeta, AccountSharedData)> = account_data_sizes
+            .iter()
+            .map(|size| create_test_account(*size))
+            .collect();
+
+        let account_refs: Vec<(&Pubkey, &AccountSharedData)> = accounts
+            .iter()
+            .map(|account| (&account.0.pubkey, &account.1))
+            .collect();
+
+        // Slot information is not used here
+        let account_data = (Slot::MAX, &account_refs[..]);
+        let hashes: Vec<_> = (0..account_data_sizes.len())
+            .map(|_| Hash::new_unique())
+            .collect();
+        let write_versions: Vec<_> = accounts
+            .iter()
+            .map(|account| account.0.write_version_obsolete)
+            .collect();
+
+        let storable_accounts =
+            StorableAccountsWithHashesAndWriteVersions::new_with_hashes_and_write_versions(
+                &account_data,
+                hashes.iter().collect(),
+                write_versions,
+            );
+
+        let temp_dir = tempdir().unwrap();
+        let tiered_storage_path = temp_dir.path().join(path_suffix);
+        let tiered_storage = TieredStorage::new_writable(tiered_storage_path, format.clone());
+        _ = tiered_storage.write_accounts(&storable_accounts, 0);
+
+        verify_hot_storage(&tiered_storage, &storable_accounts, format);
+    }
+
+    fn optional_size<T: std::cmp::PartialEq>(value: &T, default_value: T) -> usize {
+        if *value != default_value {
+            std::mem::size_of::<T>()
+        } else {
+            0
+        }
+    }
+
+    /// Verify the generated tiered storage in the test.
+    fn verify_hot_storage<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<Hash>,
+    >(
+        tiered_storage: &TieredStorage,
+        expected_accounts: &StorableAccountsWithHashesAndWriteVersions<'a, 'b, T, U, V>,
```

Comment:
The combined one also has write_versions and hashes in addition to `Vec<(StoredMeta, AccountSharedData)>`, which will also be verified in later PRs.

I'm fine changing it to `Vec<(StoredMeta, AccountSharedData)>` until we start verifying hashes and write_versions, but it will then be changed back later.

Reply:
Done!  Looks much cleaner!  I might now consider just further passing hashes and write_versions without using the StorableAccountsWithHashesAndWriteVersions.